sets/sets.go

```diff
@@ -0,0 +1,80 @@
+// Package sets is a minimal implementation of a generic set data structure.
+
+package sets
+
+import (
+	"fmt"
+	"sort"
+
+	"golang.org/x/exp/constraints"
+)
+
+// Set is a minimal set that takes only ordered types: any type that supports the
```

Comment:
Wouldn't equality be enough as a requirement?

Reply:
The reason, as you discovered below in https://github.com/Pix4D/cogito/pull/99/files#r959248227, is to be able to have a repeatable string representation.

There are other ways to skin this cat for sure (and I considered some), but this is the absolute minimum that works for the current Cogito use case.